sphinx/builders/linkcheck.py

```diff
@@ -137,10 +130,16 @@ def check_thread(self):
         if self.app.config.linkcheck_timeout:
             kwargs['timeout'] = self.app.config.linkcheck_timeout
 
+        kwargs['allow_redirects'] = True
+
         def check_uri():
             # split off anchor
             if '#' in uri:
                 req_url, anchor = uri.split('#', 1)
+                for rex in self.skip_anchors:
+                    if rex.match(uri):
```

Comment:
I was thinking in terms of allowing matching against the full URI to allow skipping based on the site name, just in the off chance that the same anchor name was reused for multiple sites but some used JS preventing checking for it.

Perhaps that's something that can be enabled with a custom flag for the regex or by setting an additional option if needed and we can default to just matching against the anchor for now?